apps/veil/src/pages/tournament/api/use-tournament-summary.ts

```diff
@@ -1,19 +1,38 @@
 import { useQuery } from '@tanstack/react-query';
-import { apiFetch } from '@/shared/utils/api-fetch';
 import { useRefetchOnNewBlock } from '@/shared/api/compact-block';
-import type { TournamentSummaryRequest, TournamentSummaryApiResponse } from '../server/summary';
+import { TournamentSummaryRequest, TournamentSummaryApiResponse } from '../server/summary';
+import { apiPostFetch } from '@/shared/utils/api-fetch';
+import { useMemo } from 'react';
+
+export const BASE_LIMIT = 10;
+export const BASE_PAGE = 1;
 
 export const useTournamentSummary = (
   params?: Partial<TournamentSummaryRequest>,
   disabled?: boolean,
 ) => {
+  const epochsKey = useMemo(
+    () => (params?.epochs ? JSON.stringify(params.epochs) : undefined),
+    [params?.epochs],
+  );
+
   const query = useQuery({
-    queryKey: ['tournament-summary'],
+    queryKey: ['tournament-summary', params?.limit, params?.page, epochsKey],
```

Comment:
```suggestion
    queryKey: ['tournament-summary', epochsKey, ...params?.limit ?? [], ...params?.page ?? []],
```

query keys can't be undefined.

Reply:
react query uses the full array as the cache key, but it's certainly fine for them to be undefined if the caller doesn't supply it. we don't need to stub this with empty values